src/shared/tpm2-util.h

```diff
@@ -59,8 +59,12 @@ struct tpm2_context {
         ESYS_CONTEXT *esys_context;
 };
 
-int tpm2_context_init(const char *device, struct tpm2_context *ret_context);
-void tpm2_context_destroy(struct tpm2_context *c);
+int tpm2_context_alloc(const char *device, struct tpm2_context **ret_context);
+struct tpm2_context *tpm2_context_free(struct tpm2_context *c);
+static inline void tpm2_context_freep(struct tpm2_context **c) {
+        if (c)
+                *c = tpm2_context_free(*c);
+}
```

Comment:
I'd move this down to the generic code. It's enough that tpm2_context_free() exists in two implementations. compiler should be smart enough to suppress the wrapper if necessary.

Reply:
also, use DEFINE_TRIVIAL_CLEANUP_FUNC()

Reply:
ack, will update, thanks